src/System.Net.Security/src/System/Net/Security/SslStreamInternal.cs

```diff
@@ -31,14 +31,12 @@ internal class SslStreamInternal
         private int _nestedRead;
         private AsyncProtocolRequest _readProtocolRequest; // cached, reusable AsyncProtocolRequest used for read operations
         private AsyncProtocolRequest _writeProtocolRequest; // cached, reusable AsyncProtocolRequest used for write operations
+        private Action<Task, object> _freeBufferAction;
```

Comment:
Yeah, I believe you are correct re overlapping.

The write buffer management here seems to present some opportunities for improvement, so ideally this will all go away when that happens.  Does that seem right to you?